singleton/controller.go

```diff
@@ -10,6 +11,12 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+const (
+	// RequeueImmediately is a constant that allows for immediate RequeueAfter when you want to run your
+	// singleton controller as hot as possible in a fast requeuing loop
+	RequeueImmediately = 1 * time.Nanosecond
```

Comment:
singleton.RequeueImmediately or controller.RequeueImediately?

This is to avoid backoff retry with Requeue: true, right?

Reply:
Figured singleton might make more sense since you probably don't want to run a standard controller that hot

Reply:
And putting it in controller might suggest that there are cases to do so

Reply:
Why wouldn't you requeue: True for a regular controller?

Reply:
I think there are plenty of cases to requeue a regular controller